### PR TITLE
Fix/mobl 583 scenedelegate optimisation

### PR DESCRIPTION
**Pull request by ketanshikharebsft**
Handled nil case for window object

### CODE REVIEW

File: BlueShift-iOS-SDK/InApps/BlueShiftNotificationModalViewController.m [hunk 1]
```diff
@@ -151,6 +151,7 @@ - (void)initializeNotificationView{
             
             CGFloat bannerBottomPadding = (bannerImagePadding && bannerImagePadding.bottom > 0) ? bannerImagePadding.bottom : 0.0;
             yPadding = yPadding + imageView.layer.frame.size.height + bannerBottomPadding;
+
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/InApps/BlueShiftNotificationModalViewController.m:154]: is this extra space needed?

File: BlueShift-iOS-SDK/InApps/BlueShiftNotificationModalViewController.m [hunk 2]
```diff
@@ -486,7 +487,12 @@ - (CGRect)positionNotificationView {
     frame.size = size;
     notificationView.autoresizingMask = UIViewAutoresizingNone;
     
-    CGSize screenSize = self.window.bounds.size;
+    CGSize screenSize = CGSizeZero;
+    if (self.window) {
+        screenSize = self.window.bounds.size;
+    } else {
+        screenSize = [[UIScreen mainScreen] bounds].size;
+    }
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/InApps/BlueShiftNotificationModalViewController.m:495]: can we move this code to a method and use it above as well? to get width and height?

File: BlueShift-iOS-SDK/InApps/BlueShiftNotificationWebViewController.m [hunk 3]
```diff
@@ -219,7 +219,13 @@ - (CGRect)positionWebView{
     frame.size = size;
     webView.autoresizingMask = UIViewAutoresizingNone;
     
-    CGSize screenSize = self.window.bounds.size;
+    CGSize screenSize = CGSizeZero;
+    if (self.window) {
+        screenSize = self.window.bounds.size;
+    } else {
+        screenSize = [[UIScreen mainScreen] bounds].size;
+    }
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/InApps/BlueShiftNotificationWebViewController.m:227]: code can be optimized if we move this code to a utility/helper method as mentioned above

File: BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationHelper.m [hunk 4]
```diff
@@ -120,6 +122,25 @@ + (UIWindow *)getApplicationKeyWindow {
...
+    }
+}
+
++ (UIEdgeInsets)getApplicationWindowSafeAreaInsets:(UIWindow*)window API_AVAILABLE(ios(11.0)) {
+    if (window) {
+        return window.safeAreaInsets;
+    } else if ([BlueShiftInAppNotificationHelper getApplicationKeyWindow]) {
+        return [BlueShiftInAppNotificationHelper getApplicationKeyWindow].safeAreaInsets;
+    }
+    return  UIEdgeInsetsZero;
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationHelper.m:141]: shall we put this inside `else` block? just for improving the readability. Like the above method

File: BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationHelper.m [hunk 5]
```diff
@@ -92,7 +92,8 @@ + (CGFloat)getPresentationAreaHeightForWindow:(UIWindow*)window {
         topMargin = [[UIApplication sharedApplication] statusBarFrame].size.height;
         bottomMargin = [[UIApplication sharedApplication] statusBarFrame].size.height;
     }
-    CGFloat presentationAreaHeight = window.bounds.size.height - topMargin - bottomMargin;
+    CGFloat windowHeight = [self getApplicationWindowSize:window].height;
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationHelper.m:95]: the window size will always be present right? do we need any nill checks? I saw the method def, just double checking.

Comment (ketanshikharebsft) [reply BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationHelper.m:95]: Yes, a valid size will be returned all the time.

File: BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationHelper.m (hunk 4 above)
Comment (ketanshikharebsft) [reply BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationHelper.m:141]: Yes, I can change it.

**Review by rahulrvp — APPROVED** [review]
LGTM

**Pull request merged**